src/main/java/org/openstreetmap/atlas/checks/validation/areas/PedestrianAreaOverlappingEdgeCheck.java

```diff
@@ -0,0 +1,245 @@
+                final SnappedLocation snappedStartLoc = edgeSegmentStartLocation
+                        .snapTo(areaPolygon);
+                // Snapped end location
+                final SnappedLocation snappedEndLoc = edgeSegmentEndLocation.snapTo(areaPolygon);
+                // Distance of original start location to the snapped location
+                final Distance startLocDistance = snappedStartLoc.getDistance();
+                // Distance of original end location to the snapped location
+                final Distance endLocDistance = snappedEndLoc.getDistance();
+                // Checks if intersection is at start of the edge
+                final boolean intersectsAtStart = locationsInSegments
+                        .contains(edgeSegmentStartLocation);
+                // Checks if intersection is at end of the edge
+                final boolean intersectsAtEnd = locationsInSegments
+                        .contains(edgeSegmentEndLocation);
+                // If both ends of the edge intersects the area, check if it is properly snapped by
+                // checking if both the intersecting points are on the same segment.
+                // If none of the filtered segments have both the start and end nodes, then add the
+                // edge and all its connected edges that are within the area to flag them.
+                if ((intersectsAtStart && intersectsAtEnd
+                        && this.isNotSnappedToSegments(segments, segment))
+                        // If any one of the end of the connected edge is fully enclosed within the
+                        // area, flag the edge and all its connected edges that are within the area.
+                        || (intersectsAtStart && !intersectsAtEnd
+                                && DISTANCE_GREATER_THAN_ONE_METER.test(endLocDistance)
+                                && areaPolygon.fullyGeometricallyEncloses(edgeSegmentEndLocation))
+                        || (intersectsAtEnd && !intersectsAtStart
+                                && DISTANCE_GREATER_THAN_ONE_METER.test(startLocDistance)
+                                && areaPolygon
+                                        .fullyGeometricallyEncloses(edgeSegmentStartLocation)))
+                {
+                    // Collect all connected edges that are within the pedestrian area
+                    edge.connectedEdges().stream().filter(connectedEdge -> this
+                            .isValidIntersectingEdge(connectedEdge, area)
+                            && areaPolygon.fullyGeometricallyEncloses(connectedEdge.asPolyLine()))
+                            .forEach(overlappingEdges::add);
+                    // Add the intersecting edge as well to the set
+                    overlappingEdges.add(edge);
+                }
+            });
+        }
+        if (!overlappingEdges.isEmpty())
+        {
+            this.markAsFlagged(object.getOsmIdentifier());
```

Comment:
Did you want to mark Atlas ids instead?

Reply:
I am marking the osm identifier of the area and atlas identifiers of the edges that it overlap.

Reply:
I see! Sorry I couldn't locate where that was happening with edges